common/app/model/content/Atoms.scala

```diff
@@ -241,6 +236,19 @@ object MediaAtom extends common.Logging {
     )
   }
 
+  def activeYouTubeAsset(atom: MediaAtom): Option[MediaAsset] = {
+    val defaultAsset = atom.assets.headOption
+
+    val activeAsset = for {
+      version <- atom.activeVersion
+      asset <- atom.assets.find(_.version == version)
+    } yield asset
+
+    val asset = activeAsset orElse defaultAsset
+
+    // sanity check the platform to avoid template rendering errors
+    asset.filter(_.platform == MediaAssetPlatform.Youtube)
+  }
```

Comment:
I was not talking about the templates. I was more wondering if we can have an `activeAssets` function in the `MediaAtom` class that would return a `Seq[MediaAsset]` rather than an `Option`, so it can be used in the `Hosted` template as well as the youtube templates (ie: `atom.activeAssets.headOption`)

It is better to have the logic in only one place. How is hosted template deal with the active version at the moment?